File: contest/src/Domain/Candidate.java
package Domain;

/**
 * Created by Sebi on 10/7/2016.
 */

/**
 * Gestioneaza obiectele tip Candidate
 */
public class Candidate implements HasId<Integer> {
    private Integer id;
    private String name;
    private String telephone;
    private String address;

    /**
     * Constructor
     * @param id - Integer
     * @param name - String
     * @param telephone - String
     * @param address - String
     */
    public Candidate(Integer id, String name, String telephone, String address){
        this.id = id;
        this.name = name;
        this.telephone = telephone;
        this.address = address;
    }

    /**
     * Getters
     */
    public Integer getId(){
        return id;
    }

    public String getName(){
        return name;
    }

    public String getTelephone(){
        return telephone;
    }

    public String getAddress(){
        return address;
    }

    /**
     * Setters
     */
    public void setId(Integer id){
        this.id = id;
    }

    public void setName(String name){
        this.name = name;
    }

    public void setTelephone(String telephone){
        this.telephone = telephone;
    }

    public void setAddress(String address){
        this.address = address;
    }

    /*
    To string method
     */
    @Override
    public String toString(){
        return id + " " + name + " " + telephone + " " + address;
    }

}

File: contestC#/contest/contest/Domain/Candidate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace contest.Domain
{
    public class Candidate : HasID<int>
    {
        //attributes
        private int id;
        private string name;
        private string telephone;
        private string address;

        //getters and setters
        public int Id
        {
            get
            {
                return id;
            }
            set
            {
                id = value;
            }
        }

        public string Name
        {
            get
            {
                return name;
            }
            set
            {
                name = value;
            }
        }

        public string Telephone
        {
            get
            {
                return telephone;
            }
            set
            {
                telephone = value;
            }
        }

        public string Address
        {
            get
            {
                return address;
            }
            set
            {
                address = value;
            }
        }

        public string IdName => $"{Id} {Name}"; //read only property
        
        //Constructor 
        public Candidate(int id, string name, string telephone, string address)
        {
            this.id = id;
            this.name = name;
            this.telephone = telephone;
            this.address = address;
        }

        //to string method 
        public override string ToString()
        {
            return id + " " + name + " " + telephone + " " + address;
        }

        //equals method
        public override bool Equals(object obj)
        {
            if (obj == null)
            {
                return false;
            }
            if (!(obj is Candidate))
            {
                return false;
            }
            Candidate c = (Candidate)obj;
            return id == c.id;
        }
    }
}
